lib/msprime.c

```diff
@@ -4267,7 +4275,8 @@ msp_run_sweep(msp_t *self)
             label = (label_id_t) j;
             recomb_mass = fenwick_get_total(&self->links[label]);
             sweep_pop_sizes[j] = avl_count(&self->populations[0].ancestors[label]);
-            rec_rates[j] = recomb_mass;
+            rec_rates[j] = self->model.model_rate_to_generation_rate(
```

Comment:
Looks like sweeps are done in generation time. Whether they should be I'm not sure but this line was a quick way of fixing the mismatch. Looks like we replaced one issue with another!

Reply:
We definitely shouldn't be doing sweeps in generations, so I think making this change makes the code more wrong. I think the least-wrong thing to do is revert, skip the offending test (which is tuned in a parameter region when the code was more wrong), and open an issue to get the test into a sensible parameter region.

Reply:
Ok, I'll revert and skip.